modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryBasicTests.java

```diff
@@ -217,10 +217,16 @@ public void testDeleteByQueryOnReadOnlyIndex() throws Exception {
         String block = randomFrom(SETTING_READ_ONLY, SETTING_READ_ONLY_ALLOW_DELETE);
         try {
             enableIndexBlock("test", block);
+            if (block.equals(SETTING_READ_ONLY_ALLOW_DELETE)) {
+                setDiskAllocationDeciderEnabled(false);
```

Comment:
I think we should randomize this such that we sometimes test with the decider enabled and sometimes with it disabled. Then verify that the the delete by query succeeds when the decider is enabled.

We should change the `cluster.info.update.interval` in the enabled case to be smaller than the default, maybe 500ms.

Reply:
The mimimum value of `cluster.info.update.interval` is 10s, so I have added some to code to refresh cluster info manually in order to speed up the test. What do you think about this?